js_modules/dagit/packages/core/src/assets/AssetsCatalogRoot.tsx

```diff
@@ -1,25 +1,71 @@
+import {gql, useQuery} from '@apollo/client';
+import {Page} from '@dagster-io/ui';
 import * as React from 'react';
 import {useParams} from 'react-router-dom';
 
+import {displayNameForAssetKey} from '../asset-graph/Utils';
+import {useDocumentTitle} from '../hooks/useDocumentTitle';
+import {Loading} from '../ui/Loading';
 import {ReloadAllButton} from '../workspace/ReloadAllButton';
 
 import {AssetPageHeader} from './AssetPageHeader';
+import {AssetView} from './AssetView';
 import {AssetsCatalogTable} from './AssetsCatalogTable';
+import {
+  AssetsCatalogRootQuery,
+  AssetsCatalogRootQueryVariables,
+} from './types/AssetsCatalogRootQuery';
 
 export const AssetsCatalogRoot = () => {
   const params = useParams();
-  const prefixPath = (params['0'] || '')
+  const currentPath: string[] = (params['0'] || '')
     .split('/')
     .filter((x: string) => x)
     .map(decodeURIComponent);
 
-  return (
-    <div>
+  const queryResult = useQuery<AssetsCatalogRootQuery, AssetsCatalogRootQueryVariables>(
+    ASSETS_CATALOG_ROOT_QUERY,
+    {
+      skip: currentPath.length === 0,
+      variables: {assetKey: {path: currentPath}},
+    },
+  );
+
+  useDocumentTitle(
+    currentPath && currentPath.length
+      ? `Assets: ${displayNameForAssetKey({path: currentPath})}`
+      : 'Assets',
+  );
+
+  return queryResult.loading ? (
+    <Page>
+      <AssetPageHeader assetKey={{path: currentPath}} />
+      <Loading queryResult={queryResult}>{() => null}</Loading>
```

Comment:
Just render a `Spinner` or something?

Reply:
Oh good find, this is a bit odd, will fix!